File: exerc-3/sort.cpp
#include <iostream>
#include <vector>
#include <string>
#include <bits/stdc++.h>

class Item {
public:
  std::string data;
  int key;
  Item(const char *_data, int _key): data { _data}, key(_key) {}
};

std::vector<Item> items = {
  {"john", 2},
  {"bill", 3},
  {"joe", 1},
  {"sara", 5},
  {"foo", 4}
};

void print_items(const std::vector<Item>items) {
  for (auto & e: items) {
    std::cout << "-->" << e.key << ": " << e.data << "\n";
  }
}

bool compareItem(Item& a, Item& b) {
  return (a.key < b.key);
}

int main(void) {
  std::cout << "--> Unordered:\n";
  print_items(items);
  std::sort(items.begin(), items.end(), compareItem);
  std::cout << "--> Sorted:\n";
  print_items(items);

  return 0;
}

File: exerc-3/sort.py
class Item:
    def __init__(self, data: str, key: int):
        self.data = data
        self.key = key

    def __repr__(self):
        return f'{self.key}: {self.data}'

Items = [ Item("john", 3), Item("joe", 2), Item("sara", 1), Item("ana", 4)]
Items.append(Item("billy", 0))

print("Unsorted:")
print(Items)

def compare_Items(a):
    return a.key

Items.sort(key=compare_Items)

print("sorting...")

print(Items)
